File: webApp/app/scripts/service/userservice.js
/**
 * 用户服务
 */
angular
    .module('homer')
    .service('UserService', function($http, CommonService) {
        var self = this;

        self.getAllUsers = function(callback) {
            var url = '/api/User';
            $http.get(url)
                .then(function success(response) {
                    if (callback) {
                        callback(response.data);
                    }
                }, function error() {
                    CommonService.httpError();
                });
        };

        self.login = function(user) {
            var url = '/api/User/login';
            return $http.post(url, user);
        };

        self.register = function(user, callback) {
            var url = '/api/User/register';
            $http.post(url, user)
                .then(function success(response) {
                    if (callback) {
                        callback(response.data);
                    }
                }, function error() {
                    CommonService.error('注册失败', '可能是您的网络连接错误或用户名重复');
                });
        };

        self.freeze = function(id, callback) {
            var url = '/api/User/freeze/' + id;
            $http.put(url)
                .then(function success(response) {
                    if (callback) {
                        callback(response.data);
                    }
                }, function error() {
                    CommonService.httpError();
                });
        };

        self.unfreeze = function(id, callback) {
            var url = '/api/User/unfreeze/' + id;
            $http.put(url)
                .then(function success(response) {
                    if (callback) {
                        callback(response.data);
                    }
                }, function error() {
                    CommonService.httpError();
                });
        };

        self.logout = function() {
            var url = '/api/User/logout';
            return $http.post(url);
        };

        self.getCurrentLoginUser = function() {
            var url = '/api/User/getCurrentLoginUser';
            return $http.get(url);
        };

        //修改个人信息
        self.update = function(id, user, callback) {
            var url = '/api/User/update/' + id;
            $http.put(url, user)
                .then(function success(response) {
                    if (callback) {
                        callback(response.data);
                    }
                }, function error() {
                    CommonService.httpError();
                });
        };

        self.pay = function(id, callback) {
            var url = '/api/User/pay/' + id;
            $http.put(url)
                .then(function success(response) {
                    if (callback) {
                        callback(response.data);
                    }
                }, function error() {
                    CommonService.httpError();
                });
        };

        return {
            pay                 : self.pay,
            login               : self.login,
            update              : self.update,
            freeze              : self.freeze,
            logout              : self.logout,
            register            : self.register,
            unfreeze            : self.unfreeze,
            getAllUsers         : self.getAllUsers,
            getCurrentLoginUser : self.getCurrentLoginUser
        };
    });

File: api/src/main/java/com/mengyunzhi/synthetical/service/UserService.java
package com.mengyunzhi.synthetical.service;

import com.mengyunzhi.synthetical.entity.User;

import javax.security.auth.message.AuthException;

/**
 * @author zhangxishuo on 2018/11/10
 * 用户逻辑
 */
public interface UserService {

    /**
     * 用户登录
     * @param user  当前登录用户
     * @return boolean 是否成功登录
     * 成功 true 失败 false
     */
    boolean login(User user);

    void register(User user);

    /**
     * 冻结用户
     */
    void freeze(Long userId);

    void unfreeze(Long userId);

    /**
     * 更新用户
     */
    void update(Long userId, User user);

    /**
     * 用户注销
     */
    void logout() throws AuthException;

    /**
     * 获取当前登录用户
     * @return  当前登录用户
     */
    User getCurrentLoginUser() throws AuthException;

    /**
     * 获取一个用户
     * @return 用户
     */
    User getOneUser();

    /**
     * 获取一个已保存的用户
     * @return 已保存的用户
     */
    User getOneSavedUser();
}
